syntax = "proto3";

// Get the successor of the peer itself
message GetSuccessor {
    string id = 1;
}

// Get the successor of the specified key 
message FindSuccessor {
    string key = 1;
}

message Successor {
    string suc_id = 1;
    bytes  suc_ip = 2;
    uint32 suc_port = 3; 
}

service Peer {
    rpc getSuccessor (GetSuccessor) returns (Successor) {}
    rpc findSuccessor (FindSuccessor) returns (Successor) {}
}
